### PR TITLE
[pickers] Clean definition of validation props

### PR DESCRIPTION
**Pull request by flaviendelangle**
Extracted from #15194

We had some duplication of the props needed to validate a date, to validate a time etc...
In #15194, I want to stop using interfaces like `UseDateFieldProps` to generate interfaces like `DatePickerFieldProps`, and I don't want to duplicate yet another time those interfaces.
So I added one interface per validator (e.g: `ExportedValidateDateProps`) that contains the props as they should be exposed by the component (picker, view, built-in field, custom field etc...). The next step would be to colocate the method to apply the default values, which are currently duplicated between fields and pickers (I did it in #14679 but I'll extract it of course).

### CODE REVIEW

File: packages/x-date-pickers-pro/src/internals/models/dateTimeRange.ts [hunk 1]
```diff
@@ -32,16 +30,8 @@ export interface UseDateTimeRangeFieldProps<
...
     RangeFieldSeparatorProps,
-    DayRangeValidationProps<TDate>,
-    TimeValidationProps<TDate>,
-    BaseDateValidationProps<TDate>,
-    DateTimeValidationProps<TDate> {
-  /**
-   * 12h/24h view for hour selection clock.
-   * @default utils.is12HourCycleInCurrentLocale()
-   */
-  ampm?: boolean;
```
Comment (flaviendelangle) [inline packages/x-date-pickers-pro/src/internals/models/dateTimeRange.ts:43]: This is not strictly related to validation but I took the opportunity to also remove duplication on this prop.

File: packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts [hunk 2]
```diff
@@ -62,8 +62,7 @@ export interface DateRangeCalendarSlotProps<TDate extends PickerValidDate>
 
 export interface ExportedDateRangeCalendarProps<TDate extends PickerValidDate>
   extends ExportedDayCalendarProps<TDate>,
-    BaseDateValidationProps<TDate>,
```
Comment (alexfauquette) [inline packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts:65]: The `BaseDateValidationProps` is still imported in this file because it's used to define `DateCalendarDefaultizedProps`

```ts
export type DateRangeCalendarDefaultizedProps<TDate extends PickerValidDate> = DefaultizedProps<
  DateRangeCalendarProps<TDate>,
  | 'views'
  | 'openTo'
  | 'reduceAnimations'
  | 'calendars'
  | 'disableDragEditing'
  | 'availableRangePositions'
  | keyof BaseDateValidationProps<TDate>
>;
```

Would it make sense to have `type ExportedValidateDateRangePropsDefaultizeKey = keyof BaseDateValidationProps<TDate>`. to completely remove links with those notion of `BaseDateValidationProps`, `BaseTimeValidationProps`, ...

Seems their is already something similar with `ExportedValidateTimeRangeProps` and `ValidateTimeRangeProps`. The second being the defaultized version of the first

**Review by alexfauquette — APPROVED** [review]
If I resume, you're introducing the following types to enforce consistency.

```ts
// Pro lib
ExportedValidateDateRangeProps
ExportedValidateTimeRangeProps
ExportedValidateDateTimeRangeProps

// MIT lib
ExportedValidateDateProps
ExportedValidateTimeProps
ExportedValidateDateTimeProps

// bonus
AmPmProps
```

I just left a comment about the notion of required props. But the rest looks good

**Comment by flaviendelangle** [comment]
> If I resume, you're introducing the following types to enforce consistency.

Exactly

File: packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts (hunk 2 above)
Comment (flaviendelangle) [reply packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.types.ts:65]: That's an interesting point.
In #14679 I'm reworking how we apply the default values (and it will remove interfaces like `DateRangeCalendarDefaultizedProps`), but I'm just moving the logic elsewhere (see https://github.com/mui/mui-x/pull/14679/files#diff-9b39de78b5e067c04513d22bde87e071d6adc77dc03839437ca392b977fc859dR42).
We could indeed introduce new interfaces next to the validator, or maybe even compare `ExportedXXX` and `XXX` and automatically list those which are optional in `ExportedXXX` but required in `XXX` :thinking: 

I'll have a look in a follow up :+1:
